lib/screens/rent_post_list.dart

```diff
@@ -0,0 +1,159 @@
+
+class RentPostList extends StatefulWidget {
+  // RentPostList({Key? key}) : super(key: key);
+
+  @override
+  _RentPostListState createState() => _RentPostListState();
+}
+
+class _RentPostListState extends State<RentPostList> {
+  bool isLoading = true;
+  var _posts = [];
+  DatabaseHelper _dbHelper;
+  @override
+  void initState() {
+    super.initState();
+    setState(() {
+      _dbHelper = DatabaseHelper.instance;
+    });
+    fetchposts();
+  }
+
+  fetchposts() async {
+    var fetchData = await _dbHelper.fetchPosts();
+
+    // for(var post in _posts){
+    //  print("postss : $_posts");
+    //  var arr = [{'key' : 1}];
+    //  print(arr[0]['key']);
+    //  print(_posts[0]['bookingMoney']);
+    // }
+    setState(() {
+      _posts = fetchData;
+      isLoading = false;
+      print(isLoading);
+    });
+  }
+
+  // final List<String> entries = <String>['Gulshan', 'Banani', 'Dhanmondi'];
+  // final List<int> colorCodes = <int>[600, 300, 100];
+  // final List<int> price = <int>[600, 300, 100];
+  List a = [
+    "https://q4g9y5a8.rocketcdn.me/wp-content/uploads/2020/02/home-banner-2020-02-min.jpg",
+    "https://pbs.twimg.com/media/Dk3BtZRW0AAov5V.jpg",
+    "https://i.pinimg.com/originals/9e/97/80/9e978052c3da56cec171517e65028965.jpg",
+  ];
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(
+        iconTheme: IconThemeData(
+            color: Colors.black), // changing the drawer icon color
+        title: Center(
+            child: Text(
+          "My Posts",
+          style: TextStyle(fontSize: 20, color: Colors.indigoAccent),
+        )),
+        backgroundColor: Colors.white70,
+      ),
+      body: isLoading == true
+          ? LoaderAnimation()
+          : Container(
+              color: Colors.blueGrey[100],
+              child: ListView.builder(
+                padding: const EdgeInsets.all(8),
+                itemCount: _posts.length,
+                itemBuilder: (BuildContext context, int index) {
+                  return Container(
+                    margin: const EdgeInsets.symmetric(
+                        vertical: 10, horizontal: 20),
+
+                    height: 350,
+                    // color: Colors.white,
+                    child: Card(
+                      // shape:Border.all(width: 5, ),
+                      // elevation: 20,
+                      // color: Colors.black,
+                      child: Column(
+                        children: <Widget>[
+                          Image.network(a[0],
+                              height: 250,
+                              width: double.infinity,
+                              fit: BoxFit.fill),
+                          Row(
+                            mainAxisAlignment: MainAxisAlignment.spaceBetween,
+                            children: <Widget>[
+                              _posts[index][colAddress] == null
+                              ? Text('')
```

Comment:
this checking should be done inside Container where you populated the data in Text widget. assigning just a text widget here will break the ui